zookeeper-contrib/zookeeper-contrib-zabtla/README.md

```diff
@@ -0,0 +1,71 @@
+# Zab-tla
+
+## Overview
+This project is devoted to providing formal specification and verification using TLA+ for the Zookeeper Atomic Broadcast(Zab) consensus protocol proposed in *Junqueira F P, Reed B C, Serafini M. Zab: High-performance broadcast for primary-backup systems[C]//2011 IEEE/IFIP 41st International Conference on Dependable Systems & Networks (DSN). IEEE, 2011: 245-256*.  
+
+We have made a formal [specification](Zab.tla) for Zab using TLA+ toolbox, and we have done a certain scale of model checking to verify the correctness of Zab.
+
+Due to the simplification of Zab algorithm description in the paper, some details in specification were modified and added. If you have any question, please let us know.
+
+You can find this document in chinese in [doc-in-chinsese](doc-in-chinese/README.md).
+
+## Requirements
+TLA+ toolbox version 1.7.0
+
+## Run
+Create specification and run models in the usual way.  
```

Comment:
I would recommend we put more descriptions on the steps to run the TLC model checker on this specification. In particular, the configurations like invariants, constraints, etc, so everyone who is interested can trivially reproduce (I see the previous discussions in this PR largely related to figure out this). What some other projects do is to also commit the toolbox configuration files (the .toolbox folder), maybe we can also consider doing the same.

Reply:
Thank you! I can write a more detailed manual  in README about how to configure TLC parameters. Since the .toolbox space is too large, I can upload some more important files in .toolbox folder(such as .settings, .projejct, Model/MC.tla, Model/MC.cfg).

Reply:
Yeah, please don't feel any pressures and take your time. I'll also take some time to comment on details of the spec this week, I think even it's not ZAB 1.0 the relevant experience are valuable when we make the 1.0 spec.

Reply:
Hi there! I have modified the project according to your requirements. Due to the insertion of Apache License in spec, I forgot to modify the line number in README (offset 17 lines backward). I will modify this minor issue in future commits. If you have any questions, please let me know. 
Next, I will take time to learn about ZAB 1.0. And if necessary, I will optimize spec from the perspective of explosive state space of model checking, and this may be a relatively long work.
By the way, since zookeeper-specifications is a new folder, Travis CI rejects this commit at the beginning of building. Could you please tell me how to pass it?(Actually I haven't touched Travis CI before)  Thank you!

Reply:
New folder structure looks awesome.

Travis CI failed because some newly introduced files missing Apache license (same reason Travis failed the first time for this pull request):
`[WARNING] Files with unapproved licenses:
  zookeeper-specifications/zab/Zab.toolbox/Model_1/MC.cfg
  zookeeper-specifications/zab/Zab.toolbox/Zab.aux
  zookeeper-specifications/zab/Zab.toolbox/Zab.tex
  zookeeper-specifications/zab/experiment/ZabWithQTest2.toolbox/ZabWithQTest2.tex
  zookeeper-specifications/zab/experiment/ZabWithQTest2.toolbox/ZabWithQTest2.aux
  zookeeper-specifications/zab/experiment/ZabWithQTest2.toolbox/Model_1/MC.cfg
  zookeeper-specifications/zab/experiment/ZabWithQTest.toolbox/ZabWithQTest.tex
  zookeeper-specifications/zab/experiment/ZabWithQTest.toolbox/ZabWithQTest.aux
  zookeeper-specifications/zab/experiment/ZabWithQTest.toolbox/Model_1/MC.cfg

[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.13:check (default-cli) on project parent: Too many files with unapproved license: 9 See RAT report in: /home/travis/build/apache/zookeeper/target/rat.txt -> [Help 1]
`

Add license header to these files should make the build pass.

To debug Travis issue - simply click the "Details" link and then go to the job page (in this case it's https://travis-ci.com/github/apache/zookeeper/jobs/506036712), and check the log for specific errors.

Reply:
Thank you! Then I know where the problem is. I just modified tla files and didn't modified other files at the same time.